src/typechecker.erl

```diff
@@ -3499,52 +3499,69 @@ type_check_function(Env, {function,_, Name, NArgs, Clauses}) ->
 %% is the singleton type. Otherwise, PatTy is none(). PatTy is a type exhausted
 %% by Pat. UBound is Ty or a subtype such that Pat :: UBound.
 add_types_pats(Pats, Tys, TEnv, VEnv, Caps) ->
-    add_types_pats(Pats, Tys, TEnv, VEnv, Caps, [], [], []).
-
-add_types_pats([], [], _TEnv, VEnv, _Caps, PatTysAcc, UBoundsAcc, CsAcc) ->
+    NewVEnv = assign_types_to_vars_bound_more_than_once(Pats, VEnv, Caps),
+    do_add_types_pats(Pats, Tys, TEnv, NewVEnv).
+
+%% Used in recursive calls from add_type_pat/4
+-spec do_add_types_pats(Pats :: [gradualizer_type:abstract_pattern()],
```

Comment:
As I understand it, this function should not be called directly, all code should go via `add_types_pats` right? Can I suggest you put a comment on `add_types_pats` to highlight this fact

Reply:
Good idea.

Btw, do we put the description above or below the spec?